services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/operations/ScanOperation.java

```diff
@@ -48,21 +59,42 @@ private ScanOperation(ScanEnhancedRequest request) {
     public ScanRequest generateRequest(TableSchema<T> tableSchema,
                                        OperationContext operationContext,
                                        DynamoDbEnhancedClientExtension extension) {
+        Map<String, AttributeValue> expressionValues = null;
+        Map<String, String> expressionNames = null;
+
+        if (this.request.filterExpression() != null) {
+            expressionValues = this.request.filterExpression().expressionValues();
+            expressionNames = this.request.filterExpression().expressionNames();
+        }
+
+        String projectionExpression = null;
+        if (!this.request.attributesToProject().isEmpty()) {
+            List<String> placeholders = new LinkedList<>();
+            Map<String, String> projectionPlaceholders = new HashMap<>();
+            for (String attr : this.request.attributesToProject()) {
+                String placeholder = PROJECTION_EXPRESSION_KEY_MAPPER.apply(attr);
+                placeholders.add(placeholder);
+                projectionPlaceholders.put(placeholder, attr);
+            }
```

Comment:
Nit: Generally speaking we've been preferring Java 8 style (stream.forEach) where it's possible to use. I don't see any reason not to use that here.

Reply:
Me, too. There was a reference to a non-final variable in a previous version of the code. I'm switching to `forEach`.